API Issues (#373)
FlutterFlow/flutterflow-documentation: docs/troubleshooting/api/client-server_errors_during_the_api_call.md

---
keywords: ['api', 'error', 'client']
slug: /client-server-errors-during-the-api-call
title: Client-Server Errors during the API call
---
# Client-Server Errors During the API Call

When calling an API in FlutterFlow, you may run into client-server errors. These typically come as status codes that indicate what went wrong, either on your end (the client) or on the server you're requesting data from.

This guide will help you understand the most common API error codes and how to fix them.

To learn more about APIs, check out our [API documentation guide](/resources/backend-logic/rest-api/).

> ## Common client-side status codes

Author: I have fixed this

---
keywords: ['api', 'error', 'client']
slug: /client-server-errors-during-the-api-call
title: Client-Server Errors during the API call
---
# Client-Server Errors During the API Call

When calling an API in FlutterFlow, you may run into client-server errors. These typically come as status codes that indicate what went wrong—either on your end (the client) or on the server you're requesting data from.

This guide will help you understand the most common API error codes and how to fix them.

> ## What is an API?

Reviewer: Remove this sub topic and add a single line about APIs such as To learn more about APIs, check out our API documentation guide and point it to our API docs. You can find it if you search for it. Please reference other guides to learn how we hyperlink internal docs.

An API (Application Programming Interface) lets two systems communicate with each other. When you make an API call, you're sending a request from your app (client) to a server, which responds with data.

## Common client-side status codes

These errors are usually caused by incorrect requests from the client side.

- **400 – Bad request**

    Your request isn’t properly formatted. This could be due to a missing parameter, a typo, or incorrect syntax.

    :::tip

>     Check the API documentation to ensure you're including the correct fields and headers.

Reviewer: ```suggestion
    Check the API's own documentation to ensure you're including the correct fields and headers.
```

    Your request was well-formed but couldn’t be processed. For example, passing a `latlng` without a comma.

- **429 – Too many requests**

    You’ve hit your rate limit. Most APIs restrict the number of requests you can make in a certain timeframe.

    :::tip
    Check your API plan limits and consider throttling requests from your app.
    :::

## Common server-side status codes

These errors occur on the API server side.

- **500 – Internal server error**

    The server ran into an unexpected issue. Double-check your request, but if it looks fine, the problem is likely on the server side.

- **501 – Not implemented**

    The HTTP method you're using (example, PUT or PATCH) isn't supported by the API yet.

- **502 – Bad gateway**

    The API server is using a gateway or proxy, and it failed to get a valid response from the upstream server. Usually a temporary issue.

- **503 – Service unavailable**

    The API server is overloaded or undergoing maintenance. Try again later.

- **504 – Gateway timeout**

    The API server took too long to respond. Try simplifying your request or reducing the size of the data being sent.

:::tip[Final tips]

> - Always check the API documentation, inspect your request, and look up error messages. If the issue persists, contact the API provider.

Reviewer: ```suggestion
- Always check the API's own documentation, inspect your request, and look up error messages. If the issue persists, contact the API provider.
```

> - Once you fix the issue, your calls should return a `200 OK`—which means everything is working as expected!

Reviewer: ```suggestion
- Once you fix the issue, your calls should return a `200 OK`, which means everything is working as expected!
```

Reviewer: Refrain from using double dashes in our documentation, please change everywhere.
